### DISCUSSION TITLE
Running on raspberry pi

### DISCUSSION THREAD

**Discussion by dmaunder** [tdorssers/TeslaPy#39] [Q&A]
Can someone please advise how I run a program using this on a raspberry pi that isnt using the gui? I dont see how the auth process is meant to work

**Comment by tdorssers** [comment]
There are two ways to do this:
1. Run teslapy on a machine with GUI and copy cache.json to your pi.
2. Create a custom authenticator function that simply prints the auth URL and asks for the redirected URL with 'void/callback' in it. You can then open the URL in a browser and paste the redirected URL into the console of your pi.

```python
def custom_auth(url):
        print('Use browser to login. Page Not Found will be shown at success.')
        print('Open this URL: ' + url)
        return input('Enter URL after authentication: ')

with teslapy.Tesla('elon@tesla.com', authenticator=custom_auth) as tesla:
	tesla.fetch_token()
```

**Reply by n-gineer** [reply]
I already have teslapy installed, opened python3, ran `import teslapy`, then all of the code in "2" above, but got the following error:
`TypeError: __init__() got an unexpected keyword argument 'authenticator'`. Is there something obvious I should have done in addition to this?

**Reply by tdorssers** [reply]
Which version do you have installed? Minimal version is 2.0.0 to support this keyword.

**Reply by n-gineer** [reply]
that's my problem:  teslapy 1.3.0 included in my latest teslausb upgrade, I'm way behind.
